src/pages/Scholarships.jsx

```diff
@@ -29,55 +29,58 @@ function ScholarshipsPage() {
   const [error, setError] = useState();
   const [sortField, setSortField] = useState('deadline');
   const [sortDir, setSortDir] = useState('asc');
-  const [amountFilterVals, setAmountFilterVals] = useState({
-    minAmount: 0,
-    maxAmount: 0,
-  });
+
   const location = useLocation();
   const history = useHistory();
 
-  useEffect(() => {
-    const params = queryString.parse(location.search, { parseNumbers: true });
-    const minParam = params.min ?? 0;
-    const maxParam = params.max ?? 0;
-    setAmountFilterVals({
-      minAmount: minParam > 0 ? minParam : 0,
-      maxAmount: maxParam > 0 ? maxParam : 0,
+  // sets min and max amount for the amount filter
+  const setMinMax = (index, val) => {
+    history.push({
+      search: queryString.stringify({
+        ...queryString.parse(location.search),
+        [index]: val,
+      }),
     });
+  };
+
+  const params = queryString.parse(location.search, { parseNumbers: true });
+  const minParam = params.minAmount;
+  const maxParam = params.maxAmount;
 
-    const newParams = { ...params };
-    if (!Number.isInteger(minParam) || minParam <= 0) delete newParams.min;
-    if (!Number.isInteger(maxParam) || maxParam <= 0) delete newParams.max;
-    history.push({ search: queryString.stringify(newParams) });
-  }, [location.search, history]);
+  if ((minParam && !Number.isInteger(minParam)) || minParam === 0) {
+    delete params.minAmount;
+    history.replace({ search: queryString.stringify(params) });
+  }
+
+  if ((maxParam && !Number.isInteger(maxParam)) || maxParam === 0) {
+    delete params.maxAmount;
+    history.replace({ search: queryString.stringify(params) });
+  }
 
   useEffect(() => {
     // TODO: Create cancellable promises
     Scholarships.list({ sortField, sortDir })
       .then((results) => {
-        const filterError =
-          amountFilterVals.maxAmount > 0 &&
-          amountFilterVals.maxAmount <= amountFilterVals.minAmount;
-        if (
-          (amountFilterVals.minAmount === 0 &&
-            amountFilterVals.maxAmount === 0) ||
-          filterError
-        )
-          setScholarships(results);
+        const filterError = maxParam > 0 && maxParam <= minParam;
```

Comment:
Hmm, I think it'd be okay to proceed with whatever amount filter values we have. There just won't be any results. Though maybe, since this case should only happen if the URL is tampered with, maybe we should prune them before creating this effect.

Reply:
Another alternative: in the effect check for min > max and if that's the case then don't bother listing scholarships and `setError` instead